DRILL-5971: Fix INT64, INT32 logical types in complex parquet reader

parthchandra opened the pull request:
Added support for the INT64 and INT32 logical types in the complex parquet reader

arina-ielchiieva commented:
@vvysotskyi please review.

vvysotskyi commented:
LGTM, +1. Except `INT_32` and `INT_64` parquet can contain other logical numeric types: `UINT_8`, `UINT_16`, `UINT_32`, `UINT_64`, `INT_8`, `INT_16`[1]. It would be great to add support for these logical types in this pull request or create another Jira for this.

[1] https://github.com/apache/parquet-format/blob/master/LogicalTypes.md#numeric-types

parthchandra commented:
Updated the PR to include all the remaining logical int types. While generating the test file for the unit test, found that some of the types were missing from the fast parquet reader so added those as well.

parthchandra commented on exec/java-exec/src/main/java/org/apache/drill/exec/store/parquet/columnreaders/ColumnReaderFactory.java:
```diff
@@ -138,6 +138,8 @@
                 return new ParquetFixedWidthDictionaryReaders.DictionaryBigIntReader(recordReader, allocateSize, descriptor, columnChunkMetaData, fixedLength, (BigIntVector) v, schemaElement);
```
OK I'll try to add these. BTW, I realized that the test files that I added for the unit tests are not annotated, so I'll need to fix those as well!

parthchandra commented:
@vvysotskyi please review - 
added fixes for date time and interval types, fixed the reading of dictionary encoded fixed width array types (including int96),  fixed object representation of interval types (used by unit tests and sqlline/jdbc), and  added a new test file generator that generates data for all (primitive) logical types in either flat or hierarchical schemas. Added those files as unit tests for the logical types.

vvysotskyi commented on exec/java-exec/src/test/java/org/apache/drill/exec/store/parquet/TestParquetComplex.java:
```diff
@@ -193,4 +200,168 @@ public void notxistsField() throws Exception {
+    };
+    testBuilder()
+        .sqlQuery(query)
+        .ordered()
```
unOrdered

vvysotskyi commented:
@parthchandra, thanks for the pull request! LGTM, +1.
